src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java

```diff
@@ -591,9 +591,28 @@ public Article readArticle(ArticleIdentifier articleIdentifier) {
   }
 
   @Override
-  public void updateIsPreprintOfDoi(ArticleIngestionIdentifier articleId, String preprintOfUrl) throws IOException {
+  public Collection<ArticleRevision> getPreprintArticlesWithoutVor() {
+    return hibernateTemplate.execute(session -> {
+      String subQuery = "" +
+          "SELECT DISTINCT fi.ingestion " +
+          "FROM ArticleFile fi " +
+          "INNER JOIN fi.ingestion ai " +
+          "WHERE fi.bucketName = 'preprints' " +
```

Comment:
Since **ArticleFile** is a fairly large table, the sub-query could take very long to perform as filtering on an the un-indexed field `bucketName` could cause a full table scan.  We may want to create an index on the `bucketName` field.